uint32 episode_iteration
bool terminate
RLEpisodeState current_episode